plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1780,7 +1793,7 @@ public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(Connector
             return TableStatisticsMetadata.empty();
         }
 
-        IcebergTableHandle tableHandle = getTableHandle(session, tableMetadata.getTable(), Optional.empty(), Optional.empty());
+        ConnectorTableHandle tableHandle = getTableHandle(session, tableMetadata.getTable(), Optional.empty(), Optional.empty());
```

Comment:
We can get a `CorruptedIcebergTableHandle` here.
Let's check whether we're dealing with a valid table handle before proceeding with the collection of statistics.